service/src/main/java/uk/gov/hmcts/reform/fpl/service/document/transformer/FurtherEvidenceDocumentsBundlesTransformer.java

```diff
@@ -4,24 +4,85 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 import uk.gov.hmcts.reform.fpl.enums.FurtherEvidenceType;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.HearingFurtherEvidenceBundle;
 import uk.gov.hmcts.reform.fpl.model.SupportingEvidenceBundle;
 import uk.gov.hmcts.reform.fpl.model.common.Element;
 import uk.gov.hmcts.reform.fpl.model.documentview.DocumentBundleView;
 import uk.gov.hmcts.reform.fpl.model.documentview.DocumentView;
+import uk.gov.hmcts.reform.fpl.model.documentview.DocumentViewType;
 
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.stream.Collectors;
 
 import static uk.gov.hmcts.reform.fpl.enums.FurtherEvidenceType.APPLICANT_STATEMENT;
+import static uk.gov.hmcts.reform.fpl.model.documentview.DocumentViewType.HMCTS;
+import static uk.gov.hmcts.reform.fpl.model.documentview.DocumentViewType.LA;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.nullSafeList;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.unwrapElements;
 
 @Component
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class FurtherEvidenceDocumentsBundlesTransformer {
 
     private final FurtherEvidenceDocumentsTransformer furtherEvidenceDocumentsTransformer;
 
-    public List<DocumentBundleView> getFurtherEvidenceDocumentBundles(
+    public List<DocumentBundleView> getFurtherEvidenceDocumentsBundleView(
+        CaseData caseData,
+        DocumentViewType view) {
+
+        List<Element<SupportingEvidenceBundle>> furtherEvidenceDocuments = new ArrayList<>();
+        List<Element<SupportingEvidenceBundle>> furtherEvidenceDocumentsLA = new ArrayList<>();
+        List<Element<SupportingEvidenceBundle>> furtherEvidenceDocumentsNC = new ArrayList<>();
+
+        List<Element<HearingFurtherEvidenceBundle>> hearingFurtherEvidenceDocuments
+            = caseData.getHearingFurtherEvidenceDocuments();
+
+        unwrapElements(hearingFurtherEvidenceDocuments).forEach(bundle -> {
+            furtherEvidenceDocuments.addAll(bundle.getSupportingEvidenceBundle());
+            furtherEvidenceDocumentsLA.addAll(bundle.getSupportingEvidenceLA());
+            furtherEvidenceDocumentsNC.addAll(bundle.getSupportingEvidenceNC());
+        });
+
+        List<DocumentBundleView> furtherEvidenceDocumentBundles = new ArrayList<>();
+
+        if (view == HMCTS) {
```

Comment:
I feel like this if statement could be simplified, what about something like
```java
        List<Element<SupportingEvidenceBundle>> furtherEvidenceDocumentBundles;

        switch (view) {
            case HMCTS:
                furtherEvidenceDocumentBundles = furtherEvidenceDocuments;
                break;
            case LA:
                furtherEvidenceDocumentBundles = furtherEvidenceDocumentsLA;
                break;
            default:
                furtherEvidenceDocumentBundles = furtherEvidenceDocumentsNC;
                break;
        }

        furtherEvidenceDocumentBundles.addAll(getFurtherEvidenceDocumentsNotLinkedToHearing(caseData, view));

        return new ArrayList<>(getFurtherEvidenceDocumentBundles(furtherEvidenceDocumentBundles));
```

Reply:
actually if we do this then move the switch into the for each then we don't even need 3 different collections, can just always use the same one e.g
```java
    public List<DocumentBundleView> getFurtherEvidenceDocumentsBundleView(CaseData caseData, DocumentViewType view) {
        List<Element<SupportingEvidenceBundle>> documentBundle = new ArrayList<>();

        unwrapElements(caseData.getHearingFurtherEvidenceDocuments()).forEach(bundle -> {
            switch (view) {
                case HMCTS:
                    documentBundle.addAll(bundle.getSupportingEvidenceBundle());
                    break;
                case LA:
                    documentBundle.addAll(bundle.getSupportingEvidenceLA());
                    break;
                case NONCONFIDENTIAL:
                    documentBundle.addAll(bundle.getSupportingEvidenceNC());
                    break;
            }
        });

        documentBundle.addAll(getFurtherEvidenceDocumentsNotLinkedToHearing(caseData, view));

        return new ArrayList<>(getFurtherEvidenceDocumentBundles(documentBundle));
    }
```

Reply:
this then improves the memory impact and performance of the system slightly as well

Reply:
yeah, it looks much simpler, thanks